docs/rfcs/component-status-reporting.md

```diff
@@ -0,0 +1,196 @@
+|     Stopping     |                                                 The component is in the process of shutting down.                                                  |
+|     Stopped      |                                                       The component has completed shutdown.                                                        |
+
+Statuses can be categorized into two groups: lifecycle and runtime.
+
+**Lifecycle Statuses**
+- Starting
+- Stopping
+- Stopped
+
+**Runtime Statuses**
+- OK
+- RecoverableError
+- FatalError
+- PermanentError 
+
+### Transitioning Between Statuses
+
+There is a finite state machine underlying the status reporting API that governs the allowable state transitions. See the state diagram below:
+
+![component-status-reporting-fsm](component-status-reporting-fsm.png)
+
+The finite state machine ensures that components progress through the lifecycle properly and it manages transitions through runtime states so that components do not need to track their state internally. Only changes in status result in new events being generated; repeat reports of the same status are ignored. PermanentError and FatalError are permanent runtime states. A component in these states cannot make any further state transitions.
+
+![component-status-reporting-example](component-status-reporting-example.png)
+
+### Automation
+
+The collector’s service implementation is responsible for starting and stopping components. Since it knows when these events occur and their outcomes, it can automate status reporting of lifecycle events for components.
+
+#### Start
+
+The collector will report a Starting event when starting a component. If an error is returned from Start, the collector will report a PermanentError event. If start returns without an error and the collector hasn't reported status itself, the collector will report an OK event.
+
+#### Shutdown
+
+The collector will report a Stopping event when shutting down a component. If Shutdown returns an error, the collector will report a PermanentError event. If Shutdown completes without an error, the collector will report a Stopped event.
+Best Practices
+
+#### Start
+
+Under most circumstances, a component does not need to report explicit status during component.Start. An exception to this rule is components that start async work (e.g. spawn a go routine). This is because async work may or may not complete before start returns and timing can vary between executions. A component can halt startup by returning an error from start. If start returns an error, automated status reporting will report a PermanentError on behalf of the component. If a component wishes to halt startup from work spawned in a go routine, it can report a FatalError. If start returns without an error automated status reporting will report OK, as long as the component hasn't already reported for itself.
```

Comment:
Is this paragraph describing a deviation?

Reply:
No it is describing the current way components report status during startup and how the collector service handles automatic status reporting.  I'll remove section once the standalone doc is merged.